internal/openfga/authz.go

```
@@ -87,6 +87,100 @@ func NewAuthz(l *logrus.Logger, cfg *util.Config) Authorization {
+
+	var tuples []client.ClientTupleKeyWithoutCondition
+	for _, tuple := range resp.Tuples {
+		userParts := strings.SplitN(tuple.Key.User, ":", 2)
+		objectParts := strings.SplitN(tuple.Key.Object, ":", 2)
+
+		for _, r := range filters {
+			if r.UserType != "" && (len(userParts) < 1 || userParts[0] != string(r.UserType)) {
+				continue
+			}
+			if r.UserId != "" && (len(userParts) < 2 || userParts[1] != string(r.UserId)) {
+				continue
+			}
+			if r.Relation != "" && tuple.Key.Relation != string(r.Relation) {
+				continue
+			}
+			if r.ObjectType != "" && (len(objectParts) < 1 || objectParts[0] != string(r.ObjectType)) {
+				continue
+			}
+			if r.ObjectId != "" && (len(objectParts) < 2 || objectParts[1] != string(r.ObjectId)) {
+				continue
+			}
+
+			// If all specified fields match for this filter, add the tuple and break to avoid duplicates
+			tuples = append(tuples, client.ClientTupleKeyWithoutCondition{
+				User:     tuple.Key.User,
+				Relation: tuple.Key.Relation,
+				Object:   tuple.Key.Object,
+			})
+			break
+		}
+	}
+	return tuples, nil
+}
+
+func (a *Authz) RemoveRelationBulk(r []RelationInput) error {
+	tuples, err := a.ListRelations(r)
+	if err != nil {
+		return err
+	}
+
+	if len(tuples) == 0 {
+		return nil
+	}
+
+	writeReq := client.ClientWriteRequest{
+		Deletes: tuples,
+	}
+	_, err = a.client.Write(context.Background()).Body(writeReq).Execute()
+	if err != nil {
+		a.logger.Errorf("OpenFGA Delete (DeleteRelations) error: %v", err)
+	} else {
+		a.logger.Infof("OpenFGA Delete (DeleteRelations): Deleted %d relations", len(tuples))
+	}
+	return err
+}
+
+func (a *Authz) UpdateRelation(r RelationInput, u RelationInput) error {
+	l := logrus.WithFields(logrus.Fields{
+		"event":            "HandleCreateAuthzRelation",
+		"user":             r.UserId,
+		"resourceId":       r.ObjectId,
+		"resourceType":     r.ObjectType,
+		"resourceRelation": r.Relation,
+	})
+
+	err := a.RemoveRelationBulk([]RelationInput{r})
+	if err != nil {
```

IMHO This code is wrong by design if RemoveRelocationBulk will return error, we will still call AddRelocation which may hide an error returning nil, so we will return nil even if some function will fail inside. Is it what we would like to have in here?

Anyway is the else part correct? I see 'Added relocation tuple....' log in the case of successful RemoveRelocation. Looks like action-based slip/copy paste error.